### PR TITLE
Feature fo v filter

### PR DESCRIPTION
**Pull request by albinnyASTAzero**
Changed from 3D to 2D rotation and changed some minor stuff.

### CODE REVIEW

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 1]
```diff
@@ -236,15 +236,28 @@ fmi2Status COSMPDummySensor::doExitInitializationMode()
...
+void rotatePoint3D(double x, double y, double z,double yaw,double pitch,double roll,double &rx,double &ry,double &rz) // TODO: Find out why 3D rotation doesn't work. Using 2D rotation instead.
 {
     double matrix[3][3];
-    double cos_yaw = cos(yaw);
-    double cos_pitch = cos(pitch);
-    double cos_roll = cos(roll);
-    double sin_yaw = sin(yaw);
-    double sin_pitch = sin(pitch);
-    double sin_roll = sin(roll);
+    double cos_yaw = cos(yaw*3.14/180);
```
Comment (oscjoha) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp]: Recommend using `M_PI` for a better approximation. It should be defined as standard.

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 2]
```diff
@@ -256,6 +269,27 @@ void rotatePoint(double x, double y, double z,double yaw,double pitch,double rol
...
+     * \param x Global relative x-coordinate.
+     * \param y Global relative y-coordinate.
+     * \param yaw Global yaw reference of rotation.
+     * \param rx Local x-coordinate after rotation. 
+     * \param ry Local y-coordinate after rotation.
+     * \param ryaw Local yaw angle to reference after rotation.
+     */
+void rotatePoint2D(double x, double y, double yaw, double &rx, double &ry, double &ryaw)
+{
+    double matrix[2][2];
```
Comment (oscjoha) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp]:
```suggestion

```

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 3]
```diff
@@ -256,6 +269,27 @@ void rotatePoint(double x, double y, double z,double yaw,double pitch,double rol
...
+    double sin_yaw = sin(yaw * 3.14/180);
+
+    rx = cos_yaw*x + sin_yaw*y;
+    ry = -sin_yaw*x + cos_yaw*y; 
+
+    ryaw = (atan2(ry,rx) * 180/3.14);
+}
+
 /*!
      * \brief Field of view filter for Low-fid sensor model. Returns true if target object is inside specified FoV.
```
Comment (oscjoha) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp:293]: briefs

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 4]
```diff
@@ -337,18 +370,20 @@ fmi2Status COSMPDummySensor::doCalc(fmi2Real currentCommunicationPoint, fmi2Real
...
                 if (veh.id().value() != ego_id.value()) {
                     // NOTE: We currently do not take sensor mounting position into account,
                     // i.e. sensor-relative coordinates are relative to center of bounding box
                     // of ego vehicle currently.
                     double trans_x = veh.base().position().x()-ego_x;
                     double trans_y = veh.base().position().y()-ego_y;
                     double trans_z = veh.base().position().z()-ego_z;
-                    double rel_x,rel_y,rel_z;
-                    rotatePoint(trans_x,trans_y,trans_z,veh.base().orientation().yaw(),veh.base().orientation().pitch(),veh.base().orientation().roll(),rel_x,rel_y,rel_z);
+                    double trans_yaw = ego_yaw;
```
Comment (oscjoha) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp]: This variable name makes no sense, an angle is not translated. 
Also it doesn't change anywhere so it's just a copy of ego_yaw?
```suggestion

```

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 5]
```diff
@@ -337,18 +370,20 @@ fmi2Status COSMPDummySensor::doCalc(fmi2Real currentCommunicationPoint, fmi2Real
...
+                    double rel_x,rel_y,rel_z,rel_yaw;
+                    rotatePoint2D(trans_x, trans_y, trans_yaw, rel_x, rel_y, rel_yaw);
```
Comment (oscjoha) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp]:
```suggestion
                    rotatePoint2D(trans_x, trans_y, ego_yaw, rel_x, rel_y, rel_yaw);

```

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 6]
```diff
@@ -367,6 +403,8 @@ fmi2Status COSMPDummySensor::doCalc(fmi2Real currentCommunicationPoint, fmi2Real
                         candidate->mutable_vehicle_classification()->CopyFrom(veh.vehicle_classification());
                         candidate->set_probability(1);
                         
+                        normal_log("OSI", "Detected vehicle! Relative Position: %f,%f,%f, Relative distance: %f, Relative Yaw: %f", rel_x, rel_y, rel_z, distance, rel_yaw);
+                        normal_log("OSI", "Unrotated - Relative Position %f,%f,%f, Relative distance: %f, Ego Yaw: %f", trans_x,trans_y,trans_z,trans_distance,trans_yaw);
```
Comment (oscjoha) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp]:
```suggestion
                        normal_log("OSI", "Unrotated - Relative Position %f,%f,%f, Relative distance: %f, Ego Yaw: %f", trans_x,trans_y,trans_z,trans_distance,ego_yaw);

```

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 7]
```diff
@@ -256,6 +269,27 @@ void rotatePoint(double x, double y, double z,double yaw,double pitch,double rol
 
 }
 
+/*!
+     * \brief 2D rotation matrix function. Calculates local X,Y coordinates and relative yaw angle from reference.  
+     * \param x Global relative x-coordinate.
```
Comment (oscjoha) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp:274]: What do you mean by "Global relative coordinates"? Global coordinates should be fixed

**Review by oscjoha — CHANGES_REQUESTED** [review]

File: examples/OSMPDummySensor/OSMPDummySensor.cpp (hunk 7 above)
Comment (albinnyASTAzero) [reply examples/OSMPDummySensor/OSMPDummySensor.cpp:274]: Its just relative coordinates unrotated. Just wanted to be extra clear that its the coordinates before the rotation has been done.

File: examples/OSMPDummySensor/OSMPDummySensor.cpp (hunk 3 above)
Comment (albinnyASTAzero) [reply examples/OSMPDummySensor/OSMPDummySensor.cpp:293]: There's already a breif there?

File: examples/OSMPDummySensor/OSMPDummySensor.cpp (hunk 1 above)
Comment (albinnyASTAzero) [reply examples/OSMPDummySensor/OSMPDummySensor.cpp]: Fixed

File: examples/OSMPDummySensor/OSMPDummySensor.cpp (hunk 3 above)
Comment (oscjoha) [reply examples/OSMPDummySensor/OSMPDummySensor.cpp:293]: row 293-298. It doesn't match the function signature

Comment (albinnyASTAzero) [reply examples/OSMPDummySensor/OSMPDummySensor.cpp:293]: My mistake. Fixed now.

**Review by oscjoha — APPROVED** [review]

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 8]
```diff
@@ -236,15 +236,28 @@ fmi2Status COSMPDummySensor::doExitInitializationMode()
...
-void rotatePoint(double x, double y, double z,double yaw,double pitch,double roll,double &rx,double &ry,double &rz)
+
+/*!
+     * \brief 3D rotation matrix function. Calculates local X,Y,Z coordinates relative from reference.  
+     * \param x Global relative x-coordinate.
+     * \param y Global relative y-coordinate.
+     * \param z Global relative z-coordinate
+     * \param yaw Global yaw reference of rotation.
+     * \param pitch Global pitch reference of rotation.
+     * \param roll Global roll reference of rotation.
```
Comment (eriadam36) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp:247]: write that input is in degrees

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 9]
```diff
@@ -256,27 +269,41 @@ void rotatePoint(double x, double y, double z,double yaw,double pitch,double rol
 
 }
 
+/*!
+     * \brief 2D rotation matrix function. Calculates local X,Y coordinates and relative yaw angle from reference.  
+     * \param x Global relative x-coordinate.
+     * \param y Global relative y-coordinate.
+     * \param yaw Global yaw reference of rotation.
```
Comment (eriadam36) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp:276]: write that input is in degrees

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 10]
```diff
@@ -236,15 +236,28 @@ fmi2Status COSMPDummySensor::doExitInitializationMode()
     return fmi2OK;
 }
 
-void rotatePoint(double x, double y, double z,double yaw,double pitch,double roll,double &rx,double &ry,double &rz)
+
+/*!
+     * \brief 3D rotation matrix function. Calculates local X,Y,Z coordinates relative from reference.
```
Comment (eriadam36) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp:241]: Can be good to write expected direction and order of the rotation here

File: examples/OSMPDummySensor/OSMPDummySensor.cpp [hunk 11]
```diff
@@ -337,18 +367,20 @@ fmi2Status COSMPDummySensor::doCalc(fmi2Real currentCommunicationPoint, fmi2Real
...
                     // NOTE: We currently do not take sensor mounting position into account,
                     // i.e. sensor-relative coordinates are relative to center of bounding box
                     // of ego vehicle currently.
                     double trans_x = veh.base().position().x()-ego_x;
                     double trans_y = veh.base().position().y()-ego_y;
                     double trans_z = veh.base().position().z()-ego_z;
-                    double rel_x,rel_y,rel_z;
-                    rotatePoint(trans_x,trans_y,trans_z,veh.base().orientation().yaw(),veh.base().orientation().pitch(),veh.base().orientation().roll(),rel_x,rel_y,rel_z);
+                    double rel_x,rel_y,rel_z,rel_yaw;
+                    rotatePoint2D(trans_x, trans_y, ego_yaw, rel_x, rel_y, rel_yaw);
```
Comment (eriadam36) [inline examples/OSMPDummySensor/OSMPDummySensor.cpp:379]: when you get 3D rotations to work change to 3D, write a TODO to change to 3D when done

**Review by eriadam36 — APPROVED** [review]

**Pull request merged**